Sources/Fullscreen/LoadingView/LoadingView.swift

```diff
@@ -12,35 +12,43 @@ import UIKit
         case success
     }
 
+    /// Decides whether the loading view displays as a full screen overlay or as a translucent HUD.
+    ///
+    /// - fullscreen: Covers the entire screen with an overlaying blocking UI elements from being accessed.
+    /// - boxed: Displays a translucent HUD that will keep things selectable beneath.
+    @objc public enum DisplayType: Int, CaseIterable {
+        case fullscreen
+        case boxed
+    }
+
     /// Allows the loading view to use a plain UIActivityIndicatorView,
     /// useful for a smooth transition between the old indicator and the new one,
     /// by using this flag we can avoid having multiple styles of showing progress in our app.
     @objc public static var shouldUseOldIndicator: Bool = false
 
     private let animationDuration: TimeInterval = 0.3
     private let loadingIndicatorSize: CGFloat = 40
+    private let boxedSize: CGFloat = 120
+
     /// Used for throttling
     private var state: State = .hidden
     private let loadingIndicatorInitialTransform: CGAffineTransform = CGAffineTransform(scaleX: 0.7, y: 0.7)
     private static let shared = LoadingView()
 
-    private lazy var newLoadingIndicator: LoadingIndicatorView = {
-        let view = LoadingIndicatorView()
-        view.translatesAutoresizingMaskIntoConstraints = false
-        view.transform = loadingIndicatorInitialTransform
-        return view
-    }()
+    private var successImageViewCenterY: NSLayoutConstraint?
+    private var loadingIndicatorCenterY: NSLayoutConstraint?
+
+    private var loadingConstraints = [NSLayoutConstraint]()
```

Comment:
This doesn't seem to be used anywhere. Is it needed?

Reply:
I was using it but then I figured I didn't need it. It's removed now :)